### PR TITLE
refactoring(glimmer): Create a clean file

### PR DESCRIPTION
**Pull request by dcyriller**
#### Description

This PR proposes to extract clean function from glimmer-printer to its file, similarly to what is done with language-js / language-html.

#### Checklist
- [ ] I’ve added tests to confirm my change works.
- [ ] (If changing the API or CLI) I’ve documented the changes I’ve made (in the `docs/` directory)
- [ ] (If the change is user-facing) I’ve added my changes to `changelog_unreleased/*/pr-XXXX.md` file following `changelog_unreleased/TEMPLATE.md`.
- [ ] I’ve read the [contributing guidelines](https://github.com/prettier/prettier/blob/master/CONTRIBUTING.md).

**✨[Try the playground for this PR](https://prettier.io/playground-redirect)✨**

### CODE REVIEW

File: src/language-handlebars/clean.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+"use strict";
+
+module.exports = function(ast, newNode) {
+  delete newNode.loc;
+  delete newNode.selfClosing;
+
+  // (Glimmer/HTML) ignore TextNode whitespace
+  if (ast.type === "TextNode") {
+    if (ast.chars.replace(/\s+/, "") === "") {
```
Comment (fisker) [inline src/language-handlebars/clean.js]: how about trim()?

File: src/language-handlebars/clean.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+      return null;
+    }
+    newNode.chars = ast.chars.replace(/^\s+/, "").replace(/\s+$/, "");
```
Comment (fisker) [inline src/language-handlebars/clean.js]: trim

File: src/language-handlebars/printer-glimmer.js [hunk 3]
```diff
@@ -622,5 +609,5 @@ function isPrettierIgnoreNode(node) {
 
 module.exports = {
   print,
-  massageAstNode: clean
+  massageAstNode: require("./clean")
```
Comment (fisker) [inline src/language-handlebars/printer-glimmer.js]: move to top?

File: src/language-handlebars/clean.js (hunk 1 above)
Comment (evilebottnawi) [reply src/language-handlebars/clean.js]: Just for you information - trim !== .replace(/\s+/, ""), trim === `/^[\s\uFEFF\xA0]+|[\s\uFEFF\xA0]+$/g, but here we can use it

Comment (dcyriller) [reply src/language-handlebars/clean.js]: good to know :)
I'll give it a try.

File: src/language-handlebars/printer-glimmer.js (hunk 3 above)
Comment (dcyriller) [reply src/language-handlebars/printer-glimmer.js]: I'm not sure

File: src/language-handlebars/clean.js (hunk 1 above)
Comment (dcyriller) [reply src/language-handlebars/clean.js]: my middle-term goal it to remove that to make it whitespace sensible (like html)

File: src/language-handlebars/printer-glimmer.js (hunk 3 above)
Comment (fisker) [reply src/language-handlebars/printer-glimmer.js]: I mean `const clean = require("./clean")` at top of this file

File: src/language-handlebars/clean.js [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+module.exports = function(ast, newNode) {
+  delete newNode.loc;
+  delete newNode.selfClosing;
+
+  // (Glimmer/HTML) ignore TextNode whitespace
+  if (ast.type === "TextNode") {
+    if (ast.chars.trim() === "") {
+      return null;
+    }
+    newNode.chars = ast.chars.trim();
```
Comment (fisker) [inline src/language-handlebars/clean.js]:
```suggestion
	const trimmed = ast.chars.trim();
    if (!trimmed) {
      return null;
    }
    newNode.chars = trimmed;
```

**Comment by dcyriller** [comment]
Thank you @thorn0

**Pull request merged**